Mathlib/RingTheory/MvPowerSeries/Inverse.lean

```diff
@@ -137,6 +138,22 @@ theorem mul_invOfUnit (φ : MvPowerSeries σ R) (u : Rˣ) (h : constantCoeff σ
         exact Nat.eq_zero_of_le_zero (H s)
 #align mv_power_series.mul_inv_of_unit MvPowerSeries.mul_invOfUnit
 
+theorem invOfUnit_mul (φ : MvPowerSeries σ R) (u : Rˣ) (h : constantCoeff σ R φ = u) :
+    invOfUnit φ u * φ = 1 := by
+  rw [← mul_cancel_right_mem_nonZeroDivisors (r := φ.invOfUnit u)]
+  rw [mul_assoc, one_mul, mul_invOfUnit _ _ h, mul_one]
+  apply mem_nonZeroDivisors_of_constantCoeff
+  simp only [constantCoeff_invOfUnit, IsUnit.mem_nonZeroDivisors (Units.isUnit u⁻¹)]
+
+theorem isUnit_iff_constantCoeff (φ : MvPowerSeries σ R) :
+    IsUnit φ ↔ IsUnit (constantCoeff σ R φ) := by
+  constructor
+  · exact IsUnit.map _
+  · rintro ⟨u, hu⟩
+    rw [isUnit_iff_exists]
+    use φ.invOfUnit u
+    simp only [mul_invOfUnit φ u hu.symm, invOfUnit_mul φ u hu.symm, true_and]
```

Comment:
```suggestion
  · intro ⟨u, hu⟩
    use ⟨_, φ.invOfUnit u, mul_invOfUnit φ u hu.symm,  invOfUnit_mul φ u hu.symm⟩
```

Reply:
You could even shrink the whole proof to avoid the constructor, but I wonder whether it is worth it.